datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1181,7 +1189,10 @@ impl SMJStream {
             let filter_columns = if chunk.buffered_batch_idx.is_some() {
                 if matches!(self.join_type, JoinType::Right) {
                     get_filter_column(&self.filter, &buffered_columns, &streamed_columns)
-                } else if matches!(self.join_type, JoinType::LeftSemi) {
+                } else if matches!(
```

Comment:
I wonder if we have to check (in the lines above) for RightAnti and RightSemi? 🤔

Reply:
I have a separate ticket for Right* joins, however I'm not sure how to build a right join on the sql level tbh. I will check it once we have Left* stabilized